apps/meteor/app/file-upload/server/lib/FileUpload.ts

```diff
@@ -464,16 +464,30 @@ export const FileUpload = {
 			return false;
 		}
 
-		if (!settings.get('FileUpload_Restrict_to_room_members') || !file?.rid) {
+		if (!file?.rid) {
 			return true;
 		}
 
-		const subscription = await Subscriptions.findOneByRoomIdAndUserId(file.rid, user._id, { projection: { _id: 1 } });
+		const fileUploadRestrictedToMembers = settings.get<boolean>('FileUpload_Restrict_to_room_members');
+		const fileUploadRestrictToUsersWhoCanAccessRoom = settings.get<boolean>('FileUpload_Restrict_to_users_who_can_access_room');
 
-		if (subscription) {
+		if (!fileUploadRestrictToUsersWhoCanAccessRoom && !fileUploadRestrictedToMembers) {
 			return true;
 		}
 
+		if (fileUploadRestrictedToMembers && !fileUploadRestrictToUsersWhoCanAccessRoom) {
+			const sub = await Subscriptions.findOneByRoomIdAndUserId(file.rid, user._id, { projection: { _id: 1 } });
+			if (!sub) {
+				return false;
+			}
+
+			return !!sub;
```

Comment:
👀

Reply:
It can be return true but idk... would it matter the extra cpu cycle to do !! 🤔

Reply:
Not that 🙈 I am not *that* critical lol

the if (!sub) is not needed since you are just returning boolean, and already doing !!sub

Reply:
which might also sound being critical, but I was confused at first, reading those lines. that's all.

Reply:
Agree with Deb, we don't need this if else, we could just return the boolean, if you need something explicit maybe a `sub !== undefined`?